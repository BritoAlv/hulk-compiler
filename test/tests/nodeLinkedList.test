type Node(value : number)
{
    value = value;
    previous = 0 as object;
    next = 0 as object;
    getValue() : number => self.value;
    setPrevious(node : Node) : Node => self.previous := node; 
    getPrevious() : Node => self.previous;
}

{
    print(("Casa" as bool) is string);
    print(("Casa" as bool) is bool);

    let node = new Node(40), previous = node.getPrevious() in 
    {
        if (previous is Node)
            print(previous.getValue())
        else
            print("Null pointer");
        
        previous := node.setPrevious(new Node(30));

        if (previous is Node)
            print(previous.getValue())
        else
            print("Null pointer");
        
        print(node.getValue());
    };
};