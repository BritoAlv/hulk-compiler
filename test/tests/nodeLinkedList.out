true
false
Null pointer
30.00000000
40.00000000

